addons/website/static/src/systray_items/website_switcher.js

```diff
@@ -11,19 +12,49 @@ const { Component } = owl;
 export class WebsiteSwitcherSystray extends Component {
     setup() {
         this.websiteService = useService('website');
+        this.notificationService = useService("notification");
+        this.actionService = useService("action");
     }
 
     getElements() {
         return this.websiteService.websites.map((website) => ({
             name: website.name,
             id: website.id,
+            domain: website.domain,
             callback: () => {
                 if (website.domain && !wUtils.isHTTPSorNakedDomainRedirection(website.domain, window.location.origin)) {
                     const { location: { pathname, search, hash } } = this.websiteService.contentWindow;
                     const path = pathname + search + hash;
                     window.location.href = `${website.domain}/web#action=website.website_preview&path=${encodeURIComponent(path)}&website_id=${encodeURIComponent(website.id)}`;
                 } else {
                     this.websiteService.goToWebsite({ websiteId: website.id });
+                    if (!website.domain) {
+                        const closeFn = this.notificationService.add(
+                            this.env._t(
+                                "This website does not have a domain configured. To avoid unexpected behaviours during website edition, we recommend closing (or refreshing) other browser tabs.\nTo remove this message please set a domain in your website settings"
+                            ),
+                            {
+                                type: "warning",
+                                title: this.env._t(
+                                    "No website domain configured for this website."
+                                ),
+                                sticky: true,
+                                buttons: [
+                                    {
+                                        onClick: () => {
+                                            this.actionService.doAction(
+                                                "website.action_website_configuration"
+                                            );
+                                            closeFn();
+                                        },
+                                        primary: true,
+                                        name: "Go to Settings",
+                                    },
+                                ],
+                            }
+                        );
+                        browser.setTimeout(closeFn, 7000);
```

Comment:
Isn't there a built-in way to tell the notificationService to close after X second without doing it ourselves?
I am saying that because testing it, I thought it was a terrible design that the toaster respect this delay even if we mouseover the toaster, it should never close itself in this case even if the delay is over: it should wait for the mouse to exit the toaster then close.
It's how toaster works usually, otherwise you get no chance to read the content of it, people are used to mouseover notification so they remain sticky and don't disappear, it's pretty much how it works everywhere (OS desktop notification, the famous "toastr" js lib etc)

But it's not related to your PR, you are obviously not going to code that case by case, it's should be a framework thing.
And bonus: when the timer is over, it should fade away slowly, so if you didn't have time to finish reading, you can move your cursor on top of it before it's fully gone and the toaster is fully shown again (revert fade) until you mouse exit it.

Reply:
Totally agree, I want the UX to be that too. Unfortunately, as of now, the notification service does not provides these features. The close delay is hardcoded as a constant (`AUTOCLOSE_DELAY`), and it seems like I can't override it easily. Hopefully it gets improved in the future 🙏🏻 . I'll keep this bit of code in mind if I see changes in the notification service!